thorlcr/activities/countproject/thcountprojectslave.cpp

```diff
@@ -167,12 +168,12 @@ class CountProjectActivity : public BaseCountProjectActivity, implements ILookAh
     }
     virtual void stop()
     {
-        PARENT::stop();
         if (first) // nextRow, therefore getPrevCount()/sendCount() never called
         {
             prevRecCount = count = getPrevCount();
             signalNext();
         }
+        PARENT::stop();
```

Comment:
A thread is in onInputFinished() waiting for a semaphore to be signaled and this does not happen because the stop() has not returned because it is itself waiting.  We need to decrement the semaphore in signalNext() before calling stop().